Description: Workaround for building the package using system jar files instead
 of precompiled ones.
Forwarded: not-needed
Author: Rail Aliev <rail@i-rs.ru>

--- a/build.xml
+++ b/build.xml
@@ -4,8 +4,8 @@
 
 	<property name="tmp.dir" value="tmp" />
 	<property name="build.dir" value="${tmp.dir}/build" />
-	<property name="build.test.dir" value="${tmp.dir}/test" />
 	<property name="dist.dir" value="dist" />
+    <property name="ext.commons.cli.lib" value="/usr/share/java/commons-cli.jar" />
 
 	<property name="jarfile-nodict" value="${tmp.dir}/bin/${build.name}-nodict-${build.ver}.jar" />
 	<property name="jarfile" value="${tmp.dir}/bin/${build.name}-${build.ver}.jar" />
@@ -14,22 +14,9 @@
 	<property name="binzip" value="${dist.dir}/${build.name}-${build.ver}.zip" />
 
 
-	<path id="compile.classpath">
-		<fileset dir="lib">
-			<include name="**/*.jar" />
-		</fileset>
-	</path>
-
-	<fileset dir="lib" id="runtime.dependencies">
-		<include name="**/*.jar" />
-		<exclude name="junit*.jar" />
-	</fileset>
-
-
 	<target name="init">
 		<tstamp />
 		<mkdir dir="${build.dir}" />
-		<mkdir dir="${build.test.dir}" />
 		<mkdir dir="${dist.dir}" />
 	</target>
 
@@ -48,7 +35,6 @@
 	<target name="compile" depends="init">
 		<javac classpath="${classpath}" destdir="${build.dir}" encoding="UTF-8" 
 				source="1.5" target="1.5">
-			<classpath refid="compile.classpath" />
 			<include name="**/*.java" />
 			<src location="src" />
 		</javac>
@@ -63,46 +49,8 @@
 
 
 
-	<target name="compile.test" depends="init,compile">
-		<javac classpath="${classpath}" destdir="${build.test.dir}" encoding="UTF-8">
-			<classpath refid="compile.classpath" />
-			<classpath location="${build.dir}" />
-
-			<src location="src-test" />
-		</javac>
-
-		<!-- Copy additional resources to the build folder. -->
-		<copy todir="${build.test.dir}">
-			<fileset dir="src-test">
-				<exclude name="**/*.java" />
-			</fileset>
-		</copy>
-	</target>
-
-
-
-	<target name="test" depends="compile, compile.test">
-		<mkdir dir="${tmp.dir}/tests" />
-		<junit printsummary="yes" haltonfailure="yes">
-			<classpath refid="compile.classpath" />
-			<classpath location="${build.dir}" />
-			<classpath location="${build.test.dir}" />
-
-			<formatter type="plain" />
-
-			<batchtest todir="${tmp.dir}/tests">
-				<fileset dir="src-test">
-					<include name="**/*Test.java" />
-				</fileset>
-			</batchtest>
-		</junit>
-	</target>
-
-
-
 	<target name="javadoc" depends="init">
 		<javadoc sourcepath="src" packagenames="morfologik.*" destdir="${tmp.dir}/javadoc">
-			<classpath refid="compile.classpath" />
 		</javadoc>
 	</target>
 
@@ -113,19 +61,12 @@
 
 		<property name="main.class" value="morfologik.tools.Launcher" />
 		
-		<pathconvert property="manifest.classpath" pathsep=" ">
-			<flattenmapper />
-			<path>
-				<fileset refid="runtime.dependencies" />
-			</path>
-		</pathconvert>
-
 		<jar jarfile="${jarfile}">
 			<fileset dir="${build.dir}" />
 			<zipfileset file="LICENSE" fullpath="META-INF/Morfologik.LICENSE" />
 			<manifest>
 				<attribute name="Main-Class" value="${main.class}" />
-				<attribute name="Class-Path" value="${manifest.classpath}" />
+                <attribute name="Class-Path" value="file://${ext.commons.cli.lib}" />
 			</manifest>
 		</jar>
 
@@ -136,7 +77,7 @@
 			<zipfileset file="LICENSE" fullpath="META-INF/Morfologik.LICENSE" />
 			<manifest>
 				<attribute name="Main-Class" value="${main.class}" />
-				<attribute name="Class-Path" value="${manifest.classpath}" />
+                <attribute name="Class-Path" value="file://${ext.commons.cli.lib}" />
 			</manifest>
 		</jar>
 	</target>
@@ -145,7 +86,6 @@
 
 	<target name="build" depends="jar">
 		<copy todir="${tmp.dir}/bin">
-			<fileset refid="runtime.dependencies" />
 			<fileset dir=".">
 				<include name="LICENSE*.txt" />
 				<include name="README.txt" />
@@ -156,7 +96,7 @@
 
 
 
-	<target name="dist" depends="clean, test, build, javadoc, licenses">
+	<target name="dist" depends="clean, build, javadoc, licenses">
 		<!-- Package source distribution -->
 		<fileset dir="." id="srcdistfiles">
 			<exclude name="tmp/**" />
